kotlin-runtime/ftl-runtime/src/main/kotlin/xyz/block/ftl/schemaextractor/ExtractSchemaRule.kt

```diff
@@ -214,44 +214,55 @@ class SchemaExtractor(
       }
 
       // Validate parameters
-      require(verb.valueParameters.size == 2) { "$verbSourcePos Verbs must have exactly two arguments, ${verb.name} did not" }
+      require(verb.valueParameters.size >= 1) { "$verbSourcePos Verbs must have at least one argument, ${verb.name} did not" }
+      require(verb.valueParameters.size <= 2) { "$verbSourcePos Verbs must have at most two arguments, ${verb.name} did not" }
       val ctxParam = verb.valueParameters.first()
-      val reqParam = verb.valueParameters.last()
+
       require(ctxParam.typeReference?.resolveType()?.fqNameOrNull()?.asString() == Context::class.qualifiedName) {
         "${verb.valueParameters.first().getLineAndColumn()} First argument of verb must be Context"
       }
 
-      require(reqParam.typeReference?.resolveType()
-        ?.let { it.toClassDescriptor().isData || it.isEmptyBuiltin() }
-        ?: false
-      ) {
-        "${verb.valueParameters.last().getLineAndColumn()} Second argument of ${verb.name} must be a data class or " +
-          "builtin.Empty"
+      if (verb.valueParameters.size == 2) {
+        val reqParam = verb.valueParameters.last()
+        require(reqParam.typeReference?.resolveType()
+          ?.let { it.toClassDescriptor().isData || it.isEmptyBuiltin() || it.isUnit() }
+          ?: false
+        ) {
+          "${verb.valueParameters.last().getLineAndColumn()} Second argument of ${verb.name} must be a data class or " +
+            "builtin.Empty"
+        }
       }
 
       // Validate return type
-      val respClass = verb.createTypeBindingForReturnType(bindingContext)?.type
-        ?: throw IllegalStateException("$verbSourcePos Could not resolve ${verb.name} return type")
-      require(respClass.toClassDescriptor().isData || respClass.isEmptyBuiltin()) {
-        "${verbSourcePos}: return type of ${verb.name} must be a data class or builtin.Empty but is ${
-          respClass.fqNameOrNull()?.asString()
-        }"
+      verb.createTypeBindingForReturnType(bindingContext)?.type?.let {
+        require(it.toClassDescriptor().isData || it.isEmptyBuiltin() || it.isUnit()) {
```

Comment:
same as above, do we want to look for a null return type instead of a `kotlin.Unit` return type?

Reply:
On this one, I think `kotlin` infers the return type of `Unit` if it's omitted.